### PR TITLE
Refactor broadcast/systembroadcast

### PR DESCRIPTION
**Pull request by nanosonde**
Get rid off redundant code and introduce addtional argument (type of broadcast) to method calls.

Also fixes this issue: https://github.com/thelsing/knx/issues/60

### CODE REVIEW

File: src/knx/data_link_layer.cpp [hunk 1]
```diff
@@ -41,11 +41,16 @@ void DataLinkLayer::dataRequest(AckType ack, AddressType addrType, uint16_t dest
...
 }
 
-void DataLinkLayer::systemBroadcastRequest(AckType ack, FrameFormat format, Priority priority, NPDU& npdu)
+void DataLinkLayer::broadcastRequest(AckType ack, FrameFormat format, Priority priority, NPDU& npdu, SystemBroadcast broadcastType)
 {
     // System Broadcast requests will always be transmitted as broadcast with KNX serial number for open media (e.g. RF medium) 
     // See 3.2.5 p.22
-    sendTelegram(npdu, ack, 0, GroupAddress, format, priority, SysBroadcast);
+#if (MEDIUM_TYPE == 5)||(MEDIUM_TYPE == 0)
+    (void)broadcastType; // not used on TP/IP, always normal broadcast
```
Comment (thelsing) [inline src/knx/data_link_layer.cpp:49]: is the (void)broadcastType; debug code or necessary?

File: src/knx/application_layer.cpp [hunk 2]
```diff
@@ -181,32 +239,36 @@ void ApplicationLayer::dataSystemBroadcastIndication(HopCountType hopType, Prior
             popByte(testInfo[1], data + 5);
             propertyId = (propertyId >> 4) & 0x0FFF;;
             testInfo[0] &= 0x0F;
-            _bau.systemNetworkParameterReadIndication(priority, hopType, objectType, propertyId, testInfo, sizeof(testInfo));
+            //TODO: _bau.systemNetworkParameterReadLocalConfirm(priority, hopType, objectType, propertyId, testInfo, sizeof(testInfo), status);
```
Comment (thelsing) [inline src/knx/application_layer.cpp:242]: Why are these calls commented out? They should have an empty standard implementation, shouldn't they?

File: src/knx/data_link_layer.cpp (hunk 1 above)
Comment (nanosonde) [reply src/knx/data_link_layer.cpp:49]: I was to lazy to put __attribute__((unused)) in these cases. Shall I use it?
It suppresses compiler warnings about unused vars.

File: src/knx/application_layer.cpp (hunk 2 above)
Comment (nanosonde) [reply src/knx/application_layer.cpp:242]: As they are not used I did not implement the empty ones. Shall I do implement the empty stuff?

Comment (nanosonde) [reply src/knx/application_layer.cpp:242]: BTW: you cannot see it really well from the DIFF.
Have a look here: https://github.com/nanosonde/knx/blob/refactor_broadcast/src/knx/application_layer.cpp#L80
And here: https://github.com/nanosonde/knx/blob/refactor_broadcast/src/knx/application_layer.cpp#L179

**Comment by nanosonde** [comment]
I just unified them to save code and to have only a single place to handle both broadcast types.
For RF and PL, some AL services are only reachable via SystemBroadcast. In contrast on TP and IP all AL services shall be reachable via Broadcast only.
For a coupler BAU this will become more important.
The coupler needs to translate between SystemBroadcast and Broadcast by looking at the APCI type in the network layer and route the message accordingly when routing between a closed and an open medium.
But even for the coupler bau with primary and secondary data link layers, it would be absolutely sufficient to have this unification and just look at the flag.

File: src/knx/application_layer.cpp (hunk 2 above)
Comment (nanosonde) [reply src/knx/application_layer.cpp]: Just added the empty confirm counterparts.

File: src/knx/data_link_layer.cpp (hunk 1 above)
Comment (nanosonde) [reply src/knx/data_link_layer.cpp]: I have removed the line for now.
I just realized that there are tons of warnings about unused vars everywhere else. :)
So maybe we should silence this warning for now as it really disturbs the compilation log essentials.

**Comment by nanosonde** [comment]
Mhmm. Thinking about it, you are right: it is closer to the specs, 
Shell we keep the seperate splitted broadcast methods?
I could just change the AppLayer code to make sure that SystemBroadcast AL service are also available for IP and TP which only use Broadcast.

**Comment by thelsing** [comment]
Then let's keep them separated and only change the application layer.

**Comment by nanosonde** [comment]
@thelsing Please review again.

**Comment by nanosonde** [comment]
@thelsing forgot the sending direction. Now everything should be in place.

**Comment by thelsing** [comment]
Seems fine to me. If we create couplers in the future we will have to change the #ifdefs to something else.

**Comment by thelsing** [comment]
Merged. Thanks.

**Pull request merged**